sdl_android/src/main/java/com/smartdevicelink/api/screen/ScreenManager.java

```diff
@@ -0,0 +1,343 @@
+package com.smartdevicelink.api.screen;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.smartdevicelink.api.BaseSubManager;
+import com.smartdevicelink.api.CompletionListener;
+import com.smartdevicelink.api.FileManager;
+import com.smartdevicelink.api.datatypes.SdlArtwork;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.rpc.SetDisplayLayout;
+import com.smartdevicelink.proxy.rpc.enums.MetadataType;
+import com.smartdevicelink.proxy.rpc.enums.PredefinedLayout;
+import com.smartdevicelink.proxy.rpc.enums.TextAlignment;
+
+import java.util.List;
+
+/**
+ * <strong>ScreenManager</strong> <br>
+ *
+ * Note: This class must be accessed through the SdlManager. Do not instantiate it by itself. <br>
+*/
+public class ScreenManager extends BaseSubManager {
+
+	private static String TAG = "ScreenManager";
+	private FileManager fileManager;
+	private SoftButtonManager softButtonManager;
+	private TextAndGraphicManager textAndGraphicManager;
+	private boolean allSubManagersFinishedUpdatingSuccesslly;
+
+	// Sub manager listener
+	private final CompletionListener subManagerListener = new CompletionListener() {
+		@Override
+		public synchronized void onComplete(boolean success) {
+			if(!success){
+				Log.d(TAG, "Sub manager failed to initialize");
```

Comment:
Should there be a check against if the submanagers failed to initialize so that the screen manager transitions to error state?

Reply:
now it sets the screen manager state to ERROR

Reply:
The start methods are used by both:
https://github.com/smartdevicelink/sdl_android/blob/feature/screen_manager/sdl_android/src/main/java/com/smartdevicelink/api/screen/ScreenManager.java#L55

Reply:
The only place that I can think of for switching to state ERROR is that when they cannot retrieve capabilities.

Reply:
The start method is called, but it is not overridden (not used) in either submanager so they will always return immediately.